### ISSUE TITLE
Impossible to login / Login.php

### ISSUE THREAD

**Issue by Biniou180** [movim/movim#170]
Bonjour Edhelas, 

First, thanks for your dedication in developping movim, I know the work it represents :-)

I've tried to install it on my RPi2 this week (Archlinux, php 7.0.5, nginx, fresh git clone on april 16th)

I'm almost there: everything seems OK, the admin panel says so, but I can't login. No message on the gui, just nothing happens when clicking on the button.
On the php logs of movim i found this line:
`PHP Warning:  Invalid argument supplied for foreach() in /srv/http/movim/app/widgets/Login/Login.php on line 243`

Could you help about this problem? (is it related to PHP 7 maybe? since arch moved to this version, I have some compatibility issues on other softs as well)

Thanks a lot!
Biniou180 / Pierre-Yves

**Comment by edhelas** [comment]
Hi,
Unfortunately this log is just a warning. Did you tried with another XMPP account (like a Movim one ?). Are you using a self-signed certificate ?

**Comment by Biniou180** [comment]
Thanks for your help!
I had not tried another XMPP account, however:
- I've tried my account on the public pod movim.eu (it works)
- I don't know how to create a movim account: the official pod (nl.movim.eu) redirect to the home page when trying to create an account.
- ...however, your question gave me the idea to test with my gmail account, it works... more or less (at least I manage to login)

Another info I forgot to give: my XMPP server is prosody 0.9.10, and installed on the same machine than the webapp + daemon.

Is there maybe a way to have more detailed logs? (a debug mode somewhere?)

**Comment by edhelas** [comment]
Hi, I've just re-added the official Movim XMPP server in the account creation list (we had a serious issue on one of our server and I had to move some services to the NL server).

If your XMPP server is on the same host than Movim it's maybe a DNS resolution issue. See if you can add your own XMPP server domain in your host file?

**Comment by Biniou180** [comment]
Ok, so I've created an account on movim.eu, tried to connect to it from my own movim instance, it works well.
I've also tried to modify my `/etc/hosts` file as requested, but i does not seems to change anything. :-/
The lines I have tried:
`my-public-ipv4   chat.mydomain.tld            myhostname`
`127.0.0.1   chat.mydomain.tld            myhostname`

(for info, a line with `mydomain.tld` is already in hosts file)

I've tried a ping just to test:

```
[root@myhostname movim]# ping chat.mydomain.tld 
PING chat.mydomain.tld (192.168.0.214) 56(84) bytes of data.
64 bytes from mydomain.tld (192.168.0.214): icmp_seq=1 ttl=64 time=0.758 ms
```

have I made anything wrong maybe?

**Comment by edhelas** [comment]
No everything seems fine for me. I have quite the same architecture than you with nl.movim.eu (Metronome XMPP server and Movim pod on the same host) and it works fine.

Did you put the XMPP SRV records as well ? I'm not sure if it will help but we can give it a try http://wiki.xmpp.org/web/SRV_Records#XMPP_SRV_records

**Comment by Biniou180** [comment]
That seems to be the problem indeed!

I didn't know a specific DNS entry was necessary for the xmpp server.
I have checked on my ovh config and I discovered the default DNS entries included an XMPP config:

```
_xmpp-client._tcp IN CNAME _xmpp-client._tcp.start.ovh.net.
_xmpp-server._tcp IN CNAME _xmpp-server._tcp.start.ovh.net.
```

Removing these rules and replacing them by my domain as explained in your link did the trick. I've still some disconnections or other bugs sometime, but at least i can go through login page.

thanks a lot! I could have gone crazy with this logless problem :-)
